tests/typing.py

```diff
@@ -2,17 +2,22 @@
 from __future__ import annotations
 
 from collections.abc import Callable, Coroutine
-from typing import Any
+from typing import TYPE_CHECKING, Any, TypeAlias
 from unittest.mock import MagicMock
 
 from aiohttp import ClientWebSocketResponse
 from aiohttp.test_utils import TestClient
 
+if TYPE_CHECKING:
+    from homeassistant.components.recorder import Recorder
+
 ClientSessionGenerator = Callable[..., Coroutine[Any, Any, TestClient]]
 MqttMockPahoClient = MagicMock
 """MagicMock for `paho.mqtt.client.Client`"""
 MqttMockHAClient = MagicMock
 """MagicMock for `homeassistant.components.mqtt.MQTT`."""
 MqttMockHAClientGenerator = Callable[..., Coroutine[Any, Any, MqttMockHAClient]]
 """MagicMock generator for `homeassistant.components.mqtt.MQTT`."""
+RecorderInstanceGenerator: TypeAlias = Callable[..., Coroutine[Any, Any, "Recorder"]]
```

Comment:
Is the tag `TypeAlias` hint needed here? So the Coroutine is returning a literaral string `"Recorder"`?

Reply:
Yes, the `TypeAlias` is needed here.

Since we only import `Recorder` under `TYPE_CHECKING`, we have to use the string `"Recorder"`.
Setting `TypeAlias` then forces type checkers to look at the content of the `"Recorder"` string.